docs/features/tuples.md

```diff
@@ -0,0 +1,227 @@
+
+A lot of details and motivation for the feature is given in 
+[https://github.com/dotnet/roslyn/issues/347](https://github.com/dotnet/roslyn/issues/347)
+
+Current design notes:
+https://github.com/dotnet/roslyn/issues/10429
+
+NOTE: The goal of this document is capture what is being implemented. As design evolves, the document will undergo adjustments. 
+Changes in design will be applied to this document as the changes are implemented.
+
+-----------------------------------
+
+Tuple types
+-----------
+
+Tuple types would be introduced with syntax very similar to a parameter list:
+
+```C#
+public (int sum, int count) Tally(IEnumerable<int> values) { ... }
+    
+var t = Tally(myValues);
+Console.WriteLine($"Sum: {t.sum}, count: {t.count}");   
+ ```
+ 
+The syntax `(int sum, int count)` indicates an anonymous struct type with public fields of the given names and types.
+
+With no further syntax additions to C#, tuple values could be created as
+``` c#
+var t1 = new (int sum, int count) (0, 1);
+var t2 = new (int sum, int count) { sum = 0, count = 0 };
```

Comment:
possibly.
And we actually have a bug here, because this currently does not work. Fails with "constructor not available" :-/  
It almost works, but something goes wrong with constructor lookup.